js/src/block-editor/controls/image.js

```diff
@@ -1,11 +1,117 @@
+/**
+ * External dependencies
+ */
+import * as React from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { MediaUpload } from '@wordpress/block-editor';
+import {
+	BaseControl,
+	Button,
+	Placeholder,
+	DropZone,
+	DropZoneProvider,
+	FormFileUpload,
+	Spinner,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
 /**
  * Internal dependencies
  */
-import { Image } from '../components';
+import { useImage } from '../hooks';
+
+const allowedTypes = [ 'image' ];
+const defaultImgId = 0;
 
 const GcbImageControl = ( props ) => {
+	const { field, getValue, instanceId, onChange } = props;
```

Comment:
`components/image.js` is moved to this file, and it's refactored to use hooks instead of HOCs.

Before, this file was just a wrapper that called `components/image.js`.